Userland/DevTools/HackStudio/HackStudioWidget.cpp

```diff
@@ -1311,8 +1311,13 @@ ErrorOr<NonnullRefPtr<GUI::Action>> HackStudioWidget::create_build_action()
 {
     auto icon = TRY(Gfx::Bitmap::load_from_file("/res/icons/16x16/build.png"sv));
     return GUI::Action::create("&Build", { Mod_Ctrl, Key_B }, icon, [this](auto&) {
-        if (warn_unsaved_changes("There are unsaved changes, do you want to save before building?") == ContinueDecision::No)
-            return;
+        if (m_auto_save_before_build_or_run) {
+            if (!save_file_changes())
+                return;
+        } else {
+            if (warn_unsaved_changes("There are unsaved changes, do you want to save before building?") == ContinueDecision::No)
+                return;
+        }
```

Comment:
If this gets any more complicated, it might be worth sharing the code between these two actions, but at this point I think it's fine.